(SUP-2909) Check that use_cached_catalog setting is false

elainemccloskey opened the pull request:
## Please check off the steps below as you complete each step
- [x] Put the Jira ticket number in parentheses in the **Title** e.g. `(SUP-XXXX) Add Super Duper State Check`
- [x] Update the Jira ticket status to `Ready for Review`
- [x] Update the Scoring Spreadsheet
- [x] Get a **review** from someone on the support team
- [x] Get a **review** from @misseuropa, @davidbastedo, or @deleon365

jordi-garcia commented:
The code looks good to me.

gavindidrichsen reviewed (approved):
Looks good to me.  Well done Elaine!

BartoszBlizniak reviewed (approved):
Looks good

misseuropa reviewed (commented):
It's good to have a link to the setting but I feel like it would be good to also tell them how to set it, if you can? (Probably can do it with a setting link like https://puppet.com/docs/puppet/7/config_set.html, but of course the rest of the team is more expert). IDK how complex it is.

deleon365 commented on README.md:
```diff
@@ -80,7 +82,7 @@ This section should be referred to for next steps when any indicator reports a f
 | S0011        | Determines if Pe-postgres Service is Running and Enabled on relevant components    | Check the Service can be started and enabled   `puppet resource service pe-postgres ensure=running`  examing /var/log/puppetlabs/postgresql/<postgresversion>/postgresql-<today>.log for failures                                                                                                         ...
 | S0012        | Determines if Puppet produced a report within the last run interval                | https://puppet.com/docs/pe/2021.4/run_puppet_on_nodes.html#troubleshooting_puppet_run_failures                                                                                                                                                                                                            ...
```
Determines when Puppet produced a report during the last run interval

1. See: https://puppet.com/docs/pe/2021.4/run_puppet_on_nodes.html#troubleshooting_puppet_run_failures                                                                                                                                                                                                                                                                                                                                                                              
2. Submit a support case.
2. Reference S0012.
3. Include the output of `puppet agent -td > debug.log 2>&1`

deleon365 commented on README.md:
```diff
@@ -80,7 +82,7 @@ This section should be referred to for next steps when any indicator reports a f
 | S0011        | Determines if Pe-postgres Service is Running and Enabled on relevant components    | Check the Service can be started and enabled   `puppet resource service pe-postgres ensure=running`  examing /var/log/puppetlabs/postgresql/<postgresversion>/postgresql-<today>.log for failures                                                                                                         ...
 | S0012        | Determines if Puppet produced a report within the last run interval                | https://puppet.com/docs/pe/2021.4/run_puppet_on_nodes.html#troubleshooting_puppet_run_failures                                                                                                                                                                                                            ...
 | S0013        | Determines if a catalog successfully applied on Puppet Agent Last run              | https://puppet.com/docs/pe/2021.4/run_puppet_on_nodes.html#troubleshooting_puppet_run_failures                                                                                                                                                                                                            ...
```
Determines when a catalog successfully applied (Del note: It's unclear to me what "applied on the puppet agent last run" means. Would customers know what this means?) on Puppet Agent Last run 

- See: https://puppet.com/docs/pe/2021.4/run_puppet_on_nodes.html#troubleshooting_puppet_run_failures                                                                                                                                                                                                                                                                                                                                                                              
- Submit a support case.
- Reference S0013.
- Include the output of  `puppet agent -td > debug.log 2>&1`

deleon365 commented on README.md:
```diff
@@ -80,7 +82,7 @@ This section should be referred to for next steps when any indicator reports a f
 | S0012        | Determines if Puppet produced a report within the last run interval                | https://puppet.com/docs/pe/2021.4/run_puppet_on_nodes.html#troubleshooting_puppet_run_failures                                                                                                                                                                                                            ...
 | S0013        | Determines if a catalog successfully applied on Puppet Agent Last run              | https://puppet.com/docs/pe/2021.4/run_puppet_on_nodes.html#troubleshooting_puppet_run_failures                                                                                                                                                                                                            ...
-
+| S0030        | Determines if Infrastructure components are running with the setting use_cached_catalog set to true                        | use_cached_catalog should not be configured on PE infrastructure nodes as it prevents the management of key infrastructure settings. Please disable this setting on any infrastructure component https://puppet.com/docs/puppet/7/configuration.html#use-cached-c ...
```
Determines when infrastructure components that run with the setting use_cached_catalog are set to true.

Don't configure use_cached_catalog on PE infrastructure nodes. It prevents the management of key infrastructure settings. 

- Disable this setting on any infrastructure component.
- See: https://puppet.com/docs/puppet/7/configuration.html#use-cached-catalog                                                                                                                                                                                                                       

Submit a support case when you encounter an error, disabling use_cached_catalog.

- Reference S0030.
- Include output errors that appeared when you attempted to change the setting.

deleon365 reviewed (changes requested):
If my suggestions aren't clear, let me know. Happy to clarify. Thanks!

MartyEwings commented on spec/acceptance/self_service_spec.rb:
```diff
@@ -15,7 +15,7 @@
     describe 'check no self_service fact is false' do
       it 'if idempotent all facts should be true' do
-        expect(host_inventory['facter']['self_service'].size).to eq(14)
+        expect(host_inventory['facter']['self_service'].size).to eq(15)
```
You will need to make this 16 now that the other PR is in place

elainemccloskey commented:
I have made the readme changes suggested to S0030, we should raise a ticket for the changes to the other tests. 
I have rebased, changed the number of tests to 16 and this should be good to merge.
